lua/lualine/component.lua

```diff
@@ -274,14 +283,22 @@ function M:draw(default_highlight, is_focused)
   if self.options.fmt then
     status = self.options.fmt(status or '')
   end
-  if type(status) == 'string' and #status > 0 then
-    self.status = status
-    self:apply_icon()
-    self:apply_padding()
-    self:apply_on_click()
-    self:apply_highlights(default_highlight)
-    self:apply_section_separators()
-    self:apply_separator()
+  local show_separators_if_empty = self.options.show_separators_if_empty
+  if type(show_separators_if_empty) == 'function' then
+    show_separators_if_empty = show_separators_if_empty()
+  end
+  if type(status) == 'string' then
+    if #status > 0 or show_separators_if_empty then
```

Comment:
Add this condition to the previous if condition. We don't need to increase the indentation level here since there's no else case.

```lua
if type(status) == 'string' and (#status > 0 or show_separators_if_empty) then
```